src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -63,28 +63,75 @@ public override string ToString()
     /// </summary>
     public class GraphSyncData
     {
+        /// <summary>
+        /// Deleted sub trees.
+        /// </summary>
         public List<Subtree> DeletedSubtrees
         {
             get;
             private set;
         }
 
+        /// <summary>
+        /// Added sub trees.
+        /// </summary>
         public List<Subtree> AddedSubtrees
         {
             get;
             private set;
         }
 
+        /// <summary>
+        /// Modified sub trees.
+        /// </summary>
         public List<Subtree> ModifiedSubtrees
         {
             get;
             private set;
         }
 
+        /// <summary>
+        /// Newly added nodes' IDs.
+        /// </summary>
+        public IEnumerable<Guid> AddedNodeIDs
+        {
+            get
+            {
+                return AddedSubtrees == null ? Enumerable.Empty<Guid>() : AddedSubtrees.Select(ts => ts.GUID);
+            }
+        }
+
```

Comment:
I'd recommend having the only constructor of `GraphSyncData` deal with `AddedSubtrees` etc. with a `null` check, so downstream operations can always count on `AddedSubtrees` being not `null`. Same for the other two subtrees. Does that make sense?